testdata/testoutput2

```diff
@@ -19612,6 +19612,939 @@ Failed: error -49 at offset 3 in replacement: unknown substring
 /a(?C)b/never_callout
 Failed: error 203 at offset 3: using callouts is disabled by the application
 
+# --------------
+
+# EXTENDED CHARACTER CLASSES (UTS#18)
+
+/[a[]/
+    [
+ 0: [
+
+/[a[]/alt_extended_class
+Failed: error 106 at offset 4: missing terminating ] for character class
+
+/[a[B]/alt_extended_class
+Failed: error 212 at offset 5: missing terminating ] for character class (note '[' must be escaped under PCRE2_ALT_EXTENDED_CLASS)
+
+/[a[B]]C/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[a]
+          cls:[B]
+          op: ||
+        ]
```

Comment:
I just realized an optimization is missing here.

PCRE2 traditionally uses a bitset for the first 256 characters of sets. Even for the new ECLASS, there should only be one bitset.

Should work like this:
- The ECLASS contains only XCLASS byte codes. 
- If, and only if an XCLASS has a bitset, then the ECLASS has a bitset.
- The ECLASS bitset is pre-computed, no extra operation is needed
- The ECLASS should be turned to CLASS / NCLASS if nothing else only the bitset is present

I know this is a complex, although important optimization, since it reduces storage and improves performance. I can do it in a follow-up patch.

Reply:
If we do full constant-folding for the CLASS/NCLASS cases, we'd not have many redundant bitsets, in practice.

I agree we could lift out the bitset, if that's important. I guess we'd need a way to fold things like `CLASS && XCLASS` and so on - remove the bitset from the XCLASS, and since the CLASS includes all chars >255 we can reduce `CLASS && XCLASS → XCLASS`.

I think I can do this fine in next week's PR, when I do the `CLASS op CLASS → CLASS` constant-folding.

Reply:
Thanks though @zherczeg, it's a great observation, and worth doing.

Reply:
If you have other priorities, I can also do these things. I don't want to put more work on you.

Reply:
Thank you.